query-engine/connector-test-kit/src/test/scala/writes/topLevelMutations/CreateMutationListSpec.scala

```diff
@@ -0,0 +1,42 @@
+package writes.topLevelMutations
+
+import org.scalatest.{FlatSpec, Matchers}
+import util.ConnectorCapability.ScalarListsCapability
+import util._
+
+class CreateMutationListSpec extends FlatSpec with Matchers with ApiSpecBase {
+
+  override def runOnlyForCapabilities = Set(ScalarListsCapability)
+
+  val schema =
+    """
+    |model User {
+    |  id      Int    @id
+    |  test    String[]
+    |}
+    |""".stripMargin
+
+  val project = ProjectDsl.fromString { schema }
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    database.setup(project)
+  }
+
+  override def beforeEach(): Unit = database.truncateProjectTables(project)
+
+  "A Create Mutation" should "should not accept null in set" in {
+    server.queryThatMustFail(
+      s"""mutation {createUser(data: {id: 1, test: {set: null}}){id, test }}""",
+      project = project,
+      errorCode = 2012
+    )
+
+    server
```

Comment:
This one could be an additional test case e.g. `should default to empty list if the parameter is not provided`.

Reply:
I think that might change once we handle the distinctions around array nullability that Postgres allows. I'll split the case when I do that.